src/utils/useStateReadState/index.ts

```diff
@@ -1,123 +1,45 @@
 import { useEffect, useState } from 'react'
-import { QueryHookOptions, useApolloClient } from '@apollo/client'
-import type { CID } from '@glif/filecoin-wallet-provider'
+import { LotusActorState } from '@glif/filecoin-actor-utils'
+import { validateAddressString } from '@glif/filecoin-address'
 
-import { Address, AddressDocument, AddressQuery } from '../../generated/graphql'
-import { decodeActorCID } from '..'
 import { useEnvironment } from '../../services/EnvironmentProvider'
 
-export type LotusRPCActorState<T> = {
-  Balance: string
-  Code: CID
-  State: T
+interface UseStateReadStateResult<T> {
+  data: LotusActorState<T> | null
+  loading: boolean
+  notFound: boolean
+  error: Error | null
 }
 
-export type MsigState<T = Address> = {
-  InitialBalance: string
-  AvailableBalance: string
-  NextTxnID: number
-  NumApprovalsThreshold: number
-  Signers: T[]
-  StartEpoch: number
-  UnlockDuration: number
-}
-
-const emptyActorState: LotusRPCActorState<any> = {
-  Balance: '0',
-  Code: { '/': '' },
-  State: null
-}
-
-/////// implemented until https://github.com/glifio/graph/issues/33
-export const useStateReadStateQuery = <T = any>(
-  baseOptions: QueryHookOptions<
-    T,
-    {
-      address: string
-    }
-  >
-): { data: LotusRPCActorState<T>; error: Error; loading: boolean } => {
-  const { lotusApi, networkName } = useEnvironment()
-  const [actorState, setActorState] = useState<LotusRPCActorState<T> | null>(
-    null
-  )
-  const [loading, setLoading] = useState(true)
-  const [error, setError] = useState<Error>(undefined)
-  const [fetchedFor, setFetchedFor] = useState<string>('')
-
-  const apolloClient = useApolloClient()
+export const useStateReadState = <T = object | null>(
```

Comment:
I think we should refactor to a `useSWR` approach here, like we should for `useMsigAvailablebalance` - comes equipped with more tools like polling, caching, and revalidation that will be useful for keeping actor states dynamically updating